### ISSUE TITLE
curses.textpad.Textbox backspace support

### ISSUE THREAD

**Issue by 570552a3-a3d2-4d8c-9df8-2fa6c65a1732** [python/cpython#60436]
BPO | [16232](https://bugs.python.org/issue16232)
--- | :---
Nosy | @ned-deily

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2012-10-14.14:38:05.797>
labels = ['extension-modules', 'easy', 'type-feature']
title = 'curses.textpad.Textbox backtrace support'
updated_at = <Date 2019-03-15.22:29:17.594>
user = 'https://bugs.python.org/emeaudroidemeaudroid'
```

bugs.python.org fields:
```python
activity = <Date 2019-03-15.22:29:17.594>
actor = 'BreamoreBoy'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Extension Modules']
creation = <Date 2012-10-14.14:38:05.797>
creator = 'emeaudroid.emeaudroid'
dependencies = []
files = []
hgrepos = []
issue_num = 16232
keywords = ['easy']
message_count = 3.0
messages = ['172888', '223108', '237423']
nosy_count = 3.0
nosy_names = ['ned.deily', 'emeaudroid.emeaudroid', 'ragreener']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'test needed'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue16232'
versions = ['Python 3.5']
```

</p></details>

### Linked PRs
* gh-103783

**Comment by k3lseytaylor** [comment]
Also running this issue (Iterm2 on Mac)
Textbox backspace is not registering and can only delete using Ctrl + H

**Comment by aidanmelen** [comment]
@k3lseytaylor I ran into the same issue. Maybe [this example](https://stackoverflow.com/a/75950899/3894599) will help.

**Comment by mblahay** [comment]
@ProfDeSI @k3lseytaylor @aidanmelen 

I am attempting to reproduce this issue, but not having any success. Please confirm which version of Python you are using.

**Comment by aidanmelen** [comment]
> @aidanmelen Do you see the issue when outside of docker?

It does not work in either of those environments. It only works when we handle `curses.ascii.DEL` key presses in addition to `curses.KEY_BACKSPACE`

**Comment by aidanmelen** [comment]
I created a PR that fixes this issue.

**Comment by mblahay** [comment]
@aidanmelen, I must be doing something different from you. What version of MacOS are you using?

**Comment by zitterbewegung** [comment]
I helped @mblahay reproduce this and in a terminal and delete was working fine in ncurses. In textedit backspace is working also but, this doesn't seem to the existing behavior.

**Comment by mblahay** [comment]
@aidanmelen We were finally able to reproduce this issue. Our problem with reproduction stemmed from our using the wrong example. When we use the example in your pull request, we also see the issue on macs (verified on three different machines). For those that are following along, the problem can be seen when running the Lib/curses/textpad.py module file as a script. It should produce a square that you can type into (and delete from!). The fix you provided does appear to work, and as far as we can tell does not cause any negative affects on Linux/Windows.

**Comment by ambv** [comment]
Fixed in Python 3.12. Thanks, Michael for reviving this and thanks Aidan for the fix! ✨ 🍰 ✨

**Comment by mblahay** [comment]
Glad I was able to help coordinate this during Pycon US 2023 sprints. Sometimes things work better in person!
